### DISCUSSION TITLE
Sliders in editor are very hard to work on

### DISCUSSION THREAD

**Discussion by Minorsonek** [ppy/osu#25118] [Ideas] [answered]
![Screenshot 2023-10-13 at 20 53 31](https://github.com/ppy/osu/assets/25911229/64745bdb-f5f4-4052-b978-d0560156f7e4)
How on earth are you supposed to create bezier/curve slider out of linear one?
Workflow is as follows:
1. You create a slider
2. Make simple straight line path (so its two anchors)
3. Then you add third anchor in the middle
4. For some reason (!) it treats it as linear anchor and the slider is angular (stable behaviour would create curve at this point already)
5. You then realize this and change curve type to bezier or curve, but to your surprise, there is no difference
6. You change ALL anchors to bezier, still no change, this slider remains linear
7. You cry and get frustrated

Can we please do better? I would expect similar behaviour to stable, or at the very least, repair the curve types so they actually work

**Comment by peppy** [comment, marked as answer]
Each time you give a node a curve type, it is a **new curve, from that point forward**. This is actually a feature, allowing much easier slider creation.

Here's what you want to do:


https://github.com/ppy/osu/assets/191335/7d8ba138-88d7-46cf-b1a3-b19ecc54d926

Once you understand the slider tool, it becomes very powerful, allowing sliders that would take hours on stable. Here's just one example:


https://github.com/ppy/osu/assets/191335/c24fa575-72a9-47d8-9d37-06d8fb9bc294

**Comment by Minorsonek** [comment]
Ok, I agree it makes sense - thanks for your response, we can close this one

However I would still believe most people would appreciate that if you put 3rd anchor to the linear slider then it becomes perfect curve by default, just like stable. But ok, maybe there are other more important things to do for the moment

**Reply by peppy** [reply]
It does if you're in placement mode. When adding nodes after placement, we may consider adding back the stable logic – it's not the first time it's come up.
